src/core/eme/session.ts

```diff
@@ -441,8 +453,16 @@ function manageSessionCreation(
 ): Observable<MediaKeys|ISessionEvent|Event> {
   return Observable.defer(() => {
     // reuse currently loaded sessions without making a new key request
-    const loadedSession = $loadedSessions.get(initData);
-    if (loadedSession && loadedSession.sessionId) {
+    const {
+      config,
+      session: loadedSession,
+    } = $loadedSessions.getConfigForInitData(initData);
+    if (
+      loadedSession &&
+      loadedSession.sessionId &&
+      config &&
+      compareMksConfigurations(config, mksConfig)
```

Comment:
So this is not a cache based on the initData anymore but both on the initData AND configuration?

Why not link the Session to its corresponding MediaKeys and having two caches:
  - MediaKeys based on the wanted key system and configuration
  - MediaKeySession based on... I don't know, initData I guess

As the configuration has nothing to do with the session but all to do with the MediaKeys instance